### PR TITLE
Return enumerable for deep_each, each_sexp, and each_of_type

### PR DESCRIPTION
**Pull request by ridiculous**
How do you feel about this?

Didn't run the tests (don't see how to do that), but they should pass, I ran em from the console.

### CODE REVIEW

**Comment by ridiculous** [comment]
@zenspider you alive?

**Comment by zenspider** [comment]
Shit. No. I dropped the ball on this one. Sec... lemme go see why.

Nope. Omnifocus is still set up correctly. I just glazed over this one. Sorry. 

The only conflict about this is my _current_ policy of still supporting 1.8 (You use `__method__`). I think that's gonna drop this year but I need to at least announce it and figure out how I'm going to reflect and version that.

**Comment by ridiculous** [comment]
Haha no prob! I can drop the use of __method__ and just reference the method name directly

**Comment by zenspider** [comment]
Yeah. go ahead and throw that at me and it'll get in w/ less thinking. :)

**Comment by ridiculous** [comment]
updated!

**Comment by ridiculous** [comment]
@zenspider :tada:

File: lib/sexp.rb [hunk 1]
```diff
@@ -108,12 +109,11 @@ def each_of_type(t, &b)
 
   ##
   # Recursively enumerates all sub-sexps skipping non-Sexp elements.
-  # TODO: test
 
   def each_sexp
+    return enum_for(:each_sexp) unless block_given?
     self.each do |sexp|
       next unless Sexp === sexp
-
```
Comment (zenspider) [inline lib/sexp.rb]: Please don't reformat my code

**Comment by zenspider** [comment]
Merged, with extra newlines. Thanks!

**Pull request closed without merging**